boa/vyper/decoder_utils.py

```diff
@@ -44,32 +46,32 @@ def __getitem__(self, subscript):
 
 
 def decode_vyper_object(mem, typ):
-    if is_bytes_m_type(typ):
+    if is_type_t(typ, BytesM_T):
```

Comment:
these all need to be changed. you can see similar case/check patterns in `vyper/codegen/core.py` and `vyper/builtins/_convert.py`